app/scripts/metamask-controller.js

```diff
@@ -4161,13 +4161,11 @@ export default class MetamaskController extends EventEmitter {
     // keyring's iframe and have the setting initialized properly
     // Optimistically called to not block MetaMask login due to
     // Ledger Keyring GitHub downtime
-    if (completedOnboarding) {
-      if (this.isKeyringAvailable(LedgerKeyring.type)) {
-        await this.withKeyringForDevice(
-          { name: HardwareDeviceNames.ledger },
-          async (keyring) => this.setLedgerTransportPreference(keyring),
-        );
-      }
+    if (completedOnboarding && this.isKeyringAvailable(LedgerKeyring.type)) {
+      this.withKeyringForDevice(
+        { name: HardwareDeviceNames.ledger },
+        async (keyring) => this.setLedgerTransportPreference(keyring),
+      );
```

Comment:
It doesn't look nice to leave this Promise dangling, but there was this comment on top:
> // This must be set as soon as possible to communicate to the
    // keyring's iframe and have the setting initialized properly
    // Optimistically called to not block MetaMask login due to
    // Ledger Keyring GitHub downtime
    
KeyringController safeguards should work despite this promise being left dangling